src/goto-instrument/contracts/utils.cpp

```diff
@@ -242,3 +242,30 @@ bool is_assigns_clause_replacement_tracking_comment(const irep_idt &comment)
   return id2string(comment).find(ASSIGNS_CLAUSE_REPLACEMENT_TRACKING) !=
          std::string::npos;
 }
+
+assignst widen_assigns(const assignst &assigns)
+{
+  assignst result;
+
+  havoc_utils_is_constantt is_constant(assigns);
+
+  for(auto it = assigns.begin(); it != assigns.end(); ++it)
```

Comment:
Could you try to use a ranged for loop instead, I don't think we strictly need the iterators here?